sympy/solvers/diophantine/diophantine.py

```diff
@@ -550,7 +550,7 @@ def solve(self, parameters=None, limit=None) -> DiophantineSolutionSet:
                 sqc = isqrt(c)
                 _c = e*sqc*D - sqa*E
                 if not _c:
-                    z = symbols("z", real=True)
+                    z = Symbol("z", real=True)
```

Comment:
Same here, why would this change be needed?

Reply:
Even if finally we decide not to add type hints to `symbols`, I still go for this change. Because
1. otherwise the following code involving `z` can not be type-checked.
2. a plural form function returns multiple results in common sense, but we only need one here
3. `symbols` has more overhead than `Symbol` as it calls `Symbol("z", real=True)` in the end

Reply:
Look at the first example in the tutorial: https://docs.sympy.org/latest/tutorial/preliminaries.html#exercises

(Then, I personally always use `Symbol` for a single symbol, but it seems like the "official" SymPy way is, or has at some stage been, to use `symbols`. So it makes more sense, as @moorepants said somewhere, to update the documentation first.)

Reply:
This should be reverted.

Reply:
> This line should be reverted.

I gave the reason for this change. What’s the reason for reverting?

Reply:
```py
Python 3.9.5 (default, Jun  4 2021, 12:28:51) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.0.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from sympy import *

In [2]: %timeit Symbol('x', real=True)
863 ns ± 6.21 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [3]: %timeit symbols('x', real=True)
2.3 µs ± 6.05 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
Only for efficiency `symbols('x')` shouldn't be endorsed inside sympy project, while users of sympy may choose between higher efficiency and more flexible syntax.

Reply:
I think it's fine to use Symbol in library code.